src/shogun/base/Parameter.cpp

```diff
@@ -3874,7 +3874,7 @@ bool TParameter::copy(TParameter* target)
 					return false;
 				}
 
-				x=x+(m_datatype.sizeof_ptype());
+				x=x+(m_datatype.sizeof_stype());
```

Comment:
If this fix is right then this code never worked for vectors of non-primitive types?